homeassistant/components/openweathermap/weather.py

```diff
@@ -83,59 +91,71 @@ def __init__(
             manufacturer=MANUFACTURER,
             name=DEFAULT_NAME,
         )
-        self._attr_supported_features = (
-            WeatherEntityFeature.FORECAST_DAILY | WeatherEntityFeature.FORECAST_HOURLY
-        )
+
+        if mode in (OWM_MODE_V30, OWM_MODE_V25):
+            self._attr_supported_features = (
+                WeatherEntityFeature.FORECAST_DAILY
+                | WeatherEntityFeature.FORECAST_HOURLY
+            )
+        elif mode == OWM_MODE_FREE_FORECAST:
+            self._attr_supported_features = WeatherEntityFeature.FORECAST_HOURLY
 
     @property
     def condition(self) -> str | None:
         """Return the current condition."""
-        return self.coordinator.data[ATTR_API_CURRENT][ATTR_API_CONDITION]
+        return self.coordinator.data[ATTR_API_CURRENT].get(ATTR_API_CONDITION)
```

Comment:
If you're tired of doing `self.coordinator.data[ATTR_API_CURRENT]`, consider adding
```py
@property
def current(self) -> dict[str, Any]:
    return self.coordinator.data[ATTR_API_CURRENT]
```
so this can be changed to
```suggestion
        return self.current.get(ATTR_API_CONDITION)
```

Reply:
This is optional btw, maybe even for a followup